packages/hawtio/src/plugins/rbac/tree-processor.ts

```diff
@@ -29,6 +29,16 @@ import { rbacService } from './rbac-service'
 export const rbacTreeProcessor: TreeProcessor = async (tree: MBeanTree) => {
   log.debug('Processing tree:', tree)
   const aclMBean = await rbacService.getACLMBean()
+
+  if (!aclMBean || aclMBean.length === 0) {
+    /*
+     * Some implementations of jolokia provision, eg. running with java -javaagent
+     * do not provide an acl mbean or implement server-side RBAC so need to skip
+     */
+    log.debug('No acl mbean available. RBAC decoration of JMX tree skipped')
+    return
+  }
+
```

Comment:
It looks good to me, but I wonder why hawtio-integartion worked well without this guard?
https://github.com/hawtio/hawtio-integration/blob/main/plugins/rbac/jmxTreeProcessor.ts#L38

Reply:
I have a theory but I'd have to check deeper. I think it might have to do with using `await` instead of `then`
If the aclMBean came undefined on a Promise.rejects for some reason instead of an error, then the code on `integration` would only run with valid values while we would be catching the undefined using await.